p2p/switch.go

```diff
@@ -286,10 +287,22 @@ func (sw *Switch) Broadcast(e Envelope) {
 //
 // NOTE: TryBroadcast uses goroutines, so order of broadcast may not be preserved.
 func (sw *Switch) TryBroadcast(e Envelope) {
-	sw.peers.ForEach(func(p Peer) {
-		go func(peer Peer) {
-			peer.TrySend(e)
-		}(p)
+	sw.Logger.Debug("TryBroadcast", "channel", e.ChannelID)
+
+	marshalMsg := e.Message
+	if wrapper, ok := e.Message.(types.Wrapper); ok {
+		marshalMsg = wrapper.Wrap()
+	}
+	marshalledMsg, err := proto.Marshal(marshalMsg)
+	if err != nil {
+		panic(fmt.Sprintf("failed to marshal proto msg %v: %v", marshalMsg, err))
+	}
+	marshalledEnvelope := MarshalledEnvelope{
```

Comment:
In my honest opinion, the best solution here is to include in the existing `p2p.Envelope` a private field with the marshalled version of the message. Everything would be simpler.